From 7f47211e73e93812def71004f0f81d6c3e7a0da6 Mon Sep 17 00:00:00 2001
From: Chris Wilson <chris@chris-wilson.co.uk>
Date: Thu, 24 Oct 2019 00:53:59 +0100
Subject: drm/i915/selftests: Flush any i915_active callback work as well
Git-commit: 7f47211e73e93812def71004f0f81d6c3e7a0da6
Patch-mainline: v5.5-rc1
References: bsc#1152489

Make trebly sure that all possible callbacks and their delayed brethren
are complete before asserting that the i915_active should be idle after
flushing all barriers.

Signed-off-by: Chris Wilson <chris@chris-wilson.co.uk>
Reviewed-by: Mika Kuoppala <mika.kuoppala@linux.intel.com>
Link: https://patchwork.freedesktop.org/patch/msgid/20191023235359.27132-1-chris@chris-wilson.co.uk
Acked-by: Thomas Zimmermann <tzimmermann@suse.de>
---
 drivers/gpu/drm/i915/gt/selftest_engine_heartbeat.c | 1 +
 1 file changed, 1 insertion(+)

diff --git a/drivers/gpu/drm/i915/gt/selftest_engine_heartbeat.c b/drivers/gpu/drm/i915/gt/selftest_engine_heartbeat.c
index 9fb5d30c43bb..768f032e6578 100644
--- a/drivers/gpu/drm/i915/gt/selftest_engine_heartbeat.c
+++ b/drivers/gpu/drm/i915/gt/selftest_engine_heartbeat.c
@@ -55,6 +55,7 @@ static void pulse_unlock_wait(struct pulse *p)
 {
 	mutex_lock(&p->active.mutex);
 	mutex_unlock(&p->active.mutex);
+	flush_work(&p->active.work);
 }
 
 static int __live_idle_pulse(struct intel_engine_cs *engine,
-- 
2.28.0

